code/modules/spells/roguetown/acolyte/necra.dm

```diff
@@ -83,6 +83,144 @@
+
+/obj/structure/underworld_portal/attack_hand(mob/living/carbon/user, list/modifiers)
+	..()
+	if(user == caster)
+		for (var/thing in contents)
+			if (istype(thing, /mob/living/carbon))
+				caster.contents.Add(src)
+				user.visible_message(
+					span_revenwarning("[user] dispels the doorway with a touch."),
+					span_purple("I close the gateway. Opening it again will release whatever is inside.")
+					)
+				return TRUE
+		qdel(src)
+		return TRUE
+	if(!do_after(user, 2 SECONDS, src))
+		return
+	gobble_mob(user, caster)
+	return TRUE
+
+/obj/structure/underworld_portal/MouseDrop_T(atom/movable/O, mob/living/user)
+	if(!isliving(O))
+		return
+	if(!istype(user) || user.incapacitated())
+		return
+	if(!Adjacent(user) || !user.Adjacent(O))
+		return
+	if(!do_after_mob(user, O, 5 SECONDS))
+		return
+	gobble_mob(O)
+	user.visible_message(
+		span_warning("[user] forces [O] into the portal!")
+	)
+	return TRUE
+
+/obj/structure/underworld_portal/proc/gobble_mob(mob/living/carbon/user, mob/living/carbon/caster)
+	if(user.mob_biotypes & MOB_UNDEAD)
+		user.visible_message(
+			span_warning("[user] is suddenly grabbed by a massive hand-and pulled through!"),
+			span_userdanger("Touching the portal, the Carriageman's hand closes around my own! No! NO!")
+			)
+		playsound(user, 'sound/misc/deadbell.ogg', 50, TRUE, -2, ignore_walls = TRUE)
+		new /obj/effect/gibspawner/generic(get_turf(user))
+		qdel(user)
+		return TRUE
+
+	user.visible_message(
+		span_revenwarning("[user] slips through the portal. Silence follows."),
+		span_purple("I touch the doorway. I slip through, and the world is silent and dark. I hear the distant rattle of a passing carriage.")
+		)
+	trapped = user
+	user.forceMove(src)
+	ADD_TRAIT(user, TRAIT_BLOODLOSS_IMMUNE, STATUS_EFFECT_TRAIT)
+	ADD_TRAIT(user, TRAIT_NOBREATH, STATUS_EFFECT_TRAIT)
+	user.add_client_colour(/datum/client_colour/monochrome)
+	time_id = addtimer(CALLBACK(src, PROC_REF(spitout_mob), user, null), 5 MINUTES, TIMER_UNIQUE | TIMER_OVERRIDE) // 5 mins timer else its spitting you out where the necran is.
+	return TRUE
+
+/obj/structure/underworld_portal/proc/spitout_mob(mob/living/carbon/user, turf/T)
+	if(!trapped)
+		return FALSE
+	if(loc in user.contents)
+		forceMove(T ? T : user.loc)
+	trapped.forceMove(loc)
+	trapped = null
+	if(time_id)
+		deltimer(time_id)
+	user.visible_message(
+		span_revenwarning("[user] slips out from the whispering portal. Shadow roils off their form like smoke."),
+		span_purple("I am pulled from Necra's realm. Air fills my lungs, my heart starts beating- I live.")
+		)
+	user.remove_client_colour(/datum/client_colour/monochrome)
+	REMOVE_TRAIT(user, TRAIT_BLOODLOSS_IMMUNE, STATUS_EFFECT_TRAIT)
+	REMOVE_TRAIT(user, TRAIT_NOBREATH, STATUS_EFFECT_TRAIT)
+	return TRUE
+
+/obj/structure/underworld_portal/container_resist(mob/living/user)
+	..()
+	if(trapped != user)
+		return
+	var/resist_prob = user.STASTR * 2.5
+	if(!prob(resist_prob))
+		return
+	spitout_mob(caster)
```

Comment:
hey after the fact but uhh..this plays the spit out message to the CASTER not the trapped person...should be maybe have been calling user or trapped here?